test/framework/integration_framework/test_mstnet_irohad.hpp

```diff
@@ -0,0 +1,114 @@
+#include "multi_sig_transactions/gossip_propagation_strategy.hpp"
+#include "multi_sig_transactions/mst_processor_impl.hpp"
+#include "multi_sig_transactions/mst_propagation_strategy_stub.hpp"
+#include "multi_sig_transactions/mst_time_provider_impl.hpp"
+#include "multi_sig_transactions/storage/mst_storage_impl.hpp"
+
+namespace integration_framework {
+
+  struct MockTransportGrpc : public iroha::network::MstTransportGrpc {
+    using MstTransportGrpc::MstTransportGrpc;
+    MOCK_METHOD2(sendState,
+                 void(const shared_model::interface::Peer &,
+                      iroha::ConstRefState));
+  };
+  /**
+   * Class for integration testing of Irohad.
+   */
+  class TestMstnetIrohad : public Irohad {
+   public:
+    TestMstnetIrohad(
+        const std::string &block_store_dir,
+        const std::string &pg_conn,
+        size_t torii_port,
+        size_t internal_port,
+        size_t max_proposal_size,
+        std::chrono::milliseconds proposal_delay,
+        std::chrono::milliseconds vote_delay,
+        const shared_model::crypto::Keypair &keypair,
+        bool is_mst_supported,
+        std::function<void(std::shared_ptr<MockTransportGrpc>)> mocker)
+        : Irohad(block_store_dir,
+                 pg_conn,
+                 torii_port,
+                 internal_port,
+                 max_proposal_size,
+                 proposal_delay,
+                 vote_delay,
+                 keypair,
+                 is_mst_supported),
+          mocker(mocker) {}
+
+    auto &getCommandServiceTransport() {
+      return command_service_transport;
+    }
+
+    void initMstProcessor() override {
+      auto mock =
+          std::make_shared<MockTransportGrpc>(async_call_,
+                                              common_objects_factory_,
+                                              transaction_factory,
+                                              batch_parser,
+                                              transaction_batch_factory_);
+      mocker(mock);
+      mst_transport = mock;
+      auto mst_completer = std::make_shared<iroha::DefaultCompleter>();
+      auto mst_storage =
+          std::make_shared<iroha::MstStorageStateImpl>(mst_completer);
+      std::shared_ptr<iroha::PropagationStrategy> mst_propagation;
+      if (is_mst_supported_) {
+        mst_propagation = std::make_shared<iroha::GossipPropagationStrategy>(
+            storage,
+            std::chrono::seconds(5) /*emitting period*/,
+            2 /*amount per once*/);
+      } else {
+        mst_propagation = std::make_shared<iroha::PropagationStrategyStub>();
+      }
+
+      auto mst_time = std::make_shared<iroha::MstTimeProviderImpl>();
+      auto fair_mst_processor = std::make_shared<iroha::FairMstProcessor>(
+          mst_transport, mst_storage, mst_propagation, mst_time);
+      mst_processor = fair_mst_processor;
+      mst_transport->subscribe(fair_mst_processor);
+      log_->info("[Init] => MST processor");
```

Comment:
Everything after `mst_transport = mock;` is the same as in real Irohad, I assume. Maybe it's possible to put this body into a static method, which will be shared among both real and this Irohades?